### PR TITLE
Initialise on load instead of 'plugins_loaded' when loaded after 'plugins_loaded'

### LINKED ISSUE #359: action-scheduler doesn't load properly unless I change 'plugins_loaded' to 'after_setup_theme' in action-scheduler.php
I've been using this library in my theme, and if I include it as is, `action_scheduler_register_2_dot_2_dot_5` and `action_scheduler_initialize_2_dot_2_dot_5` in `action-scheduler.php` are never executed. Unless I change `plugins_loaded` to `after_setup_theme`.

https://github.com/Prospress/action-scheduler/blob/master/action-scheduler.php

I've installed it using Composer, and included `action-scheduler.php` at the top of the file where the action is initialized.

I could always just make a fork but I'm wondering if this is an actual issue or it seems like I'm doing something wrong?

### PR DESCRIPTION
**Pull request by thenbrent**
To fix usage within themes, which are loaded after 'plugins_loaded'.

This is a potentially dangerous breaking change, because 3rd party code may have expected Action Scheduler to be initialized on `'plugins_loaded'` with priority > `1`. There's no way to support use in themes, and fix #359, without introducing that change. If we see issues arise during the RC period, or even after public release of 3.0.0, I propose we revert this change and document that theme usage is not supported (at least, not without some other code).

Fixes #359.

---

I've also updated the doc on **Load Order** to reflect these changes. That doc appears to have been incorrect previously, stating:

> Action Scheduler will later initialize itself on `'init'` with priority `1`

When really, Action Scheduler was initialized on `'plugins_loaded'` with priority `1`. I checked the Git history, and can't see any history of Action Scheduler initializing itself on `'init'` even [before version 1.0.0](https://github.com/woocommerce/action-scheduler/commit/2af908e1a6727846a8d11e92ab5f135c4080e120#diff-4f18dbf952a597f7ef96664aaa47df23), so I think this must have been a mistake.

### CODE REVIEW

**Comment by james-allan** [comment]
As an alternative could we check if `did_action( 'plugins_loaded' ) && ! class_exists( 'ActionScheduler' )` and if so init AS immediatly? 

The result of that would be: 

1. If it's packaged in a **plugin**, it will be loaded on `plugins_loaded`. Like it has in the past.
1. If it's packaged in a **theme**, it will be loaded when the theme includes it. A JIT of sorts.

One consequence of that would be that if there are plugin versions of AS registered, only the latest version of AS included in plugins would be loaded. The theme wouldn't have had a chance to register its version by the time plugins have loaded it.

It means themes could use AS but their version would only be used if no other plugins are loading it; plugin instances would take precedence.

It's not great but that might be a compromise we're willing to make. Not sure. 🤷‍♂

**Comment by rrennick** [comment]
> As an alternative could we check if did_action( 'plugins_loaded' ) && ! class_exists( 'ActionScheduler' ) and if so init AS immediatly?

That's an idea. This would work

```
if ( ! did_action( 'plugins_loaded' ) {
add_action( 'plugins_loaded', 'action_scheduler_register_3_dot_0_dot_0_beta_1', 0, 0 );
} else {
action_scheduler_register_3_dot_0_dot_0_beta_1();
}
```

The same logic would apply to https://github.com/woocommerce/action-scheduler/blob/master/action-scheduler.php#L32 .

**Comment by thenbrent** [comment]
I've pushed up a different approach here that follows the line of thinking above to check if `did_action( 'plugins_loaded' )` in order to maintain backward compatibility with prior versions.

Importantly, the previously proposed approach has been reverted, and that can never be used because of the issues mentioned on: https://github.com/woocommerce/action-scheduler/pull/372/commits/fc4e69362bcb2bd92cd805e10bd1dafa456e187f

> Using `'init'` means that older versions included in plugins will be initialized instead of the latest version if they are loaded by WordPress first, or more specifically, if the load the `action-scheduler.php` file first, because then their call will execute the `if ( ! class_exists( 'ActionScheduler_Versions' ) )` block attaching `ActionScheduler_Versions::initialize_latest_version()` to `'plugins_loaded'` instead of `'init'`.

File: docs/usage.md [hunk 1]
```diff
@@ -121,3 +121,7 @@ Action Scheduler will register its version on `'plugins_loaded'` with priority `
 It is recommended to load it _when the file including it is included_. However, if you need to load it on a hook, then the hook must occur before `'plugins_loaded'`, or you can use `'plugins_loaded'` with negative priority, like `-10`.
 
 Action Scheduler will later initialize itself on `'init'` with priority `1`.  Action Scheduler APIs should not be used until after `'init'` with priority `1`.
+
+### Usage in Themes
+
+When using Action Scheduler in themes, it's important to note that the latest version of Action Scheduler registered by a plugin will be used rather than the version included in the theme. This is because of the version dependency handling code using `'plugins_loaded'` since version 1.0.
```
Comment (rrennick) [inline docs/usage.md]: Wording change suggestion

> it's important to note that if Action Scheduler has been registered by a plugin then the latest version registered by a plugin will be used rather ...

**Comment by rrennick** [comment]
@thenbrent I really like that approach. Plugins embedded in themes often do not get updated. This approach allows for easy update whether or not the theme is updated.

File: docs/usage.md (hunk 1 above)
Comment (thenbrent) [reply docs/usage.md]: Changed to that and rebased in the change. 👍

**Pull request merged**